api/api.go

```diff
@@ -5,37 +5,32 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/url"
-	"time"
 
 	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/rhttp"
 )
 
 var (
 	maxRetries = 5
 )
 
 func callAPI(
-	client *http.Client,
+	client *rhttp.Client,
 	method, path, authToken string,
-	maxRetries, attempts int,
-	payload interface{}) ([]byte, int, error) {
+	payload interface{}) ([]byte, int, error,
+) {
 
-	logData := log.Data{"url": path, "method": method, "attempts": attempts}
+	logData := log.Data{"url": path, "method": method}
 
-	if attempts == 0 {
-		URL, err := url.Parse(path)
-		if err != nil {
-			log.ErrorC("Failed to create url for ImportAPI call", err, logData)
-			return nil, 0, err
-		}
-		path = URL.String()
-		logData["url"] = path
-	} else {
-		// TODO improve:  exponential backoff
-		time.Sleep(time.Duration(attempts) * 10 * time.Second)
+	URL, err := url.Parse(path)
+	if err != nil {
+		log.ErrorC("Failed to create url for API call", err, logData)
+		return nil, 0, err
 	}
+	path = URL.String()
```

Comment:
'path' is passed in as a function parameter, does it need to be shadowed here or can we just use the original value? If there is a reason consider adding a comment to highlight why.

Reply:
this is a legacy of removing the retry code, will tidy

Reply:
(or so I thought)

No, `URL` is created a few lines above, from parsing `path` (used to canonicalise/sanitise), so this is just stringification